### PR TITLE
Line dragging issue resolve

### LINKED ISSUE #21: Draggabaility of a Line does not work reliably
## Current
If the first item added to an image by a user is a line, the line remains undraggable until a textarea element is also added to the image.

## Expected
Dragging a line should be reliable and smooth.

### PR DESCRIPTION
**Pull request by Shapokhan**
fixes #21

### CODE REVIEW

File: src/Feedback.js [hunk 1]
```diff
@@ -88,6 +88,12 @@ export default function Feedback(argument) {
   const [selectedPointer, setSelectedPointer] = useState("Textarea");
   const imageParentRef = useRef(null);
   const imageScrollOwnerRef = useRef(null);
+  const [lineDragging, setLineDragging] = useState(false);
+  const [diffInInitialLeft, setDiffInInitialLeft] = useState(0);
```
Comment (omeryousaf) [inline src/Feedback.js:92]: Let's rename it from `diffInInitialLeft` to `diffInStartX`

File: src/Feedback.js [hunk 2]
```diff
@@ -152,15 +158,15 @@ export default function Feedback(argument) {
...
-    const result = commentElements.map((el) =>
-      el.id === elID
-        ? { ...el, initialLeftOffset: updatedInitialLeft, initialTopOffset: updatedInitialTop,finalLeftOffset: updatedFinalLeft, finalTopOffset: updatedFinalTop}
-        : { ...el }
-    );
-    setCommentElements(result);
-  };
+const onLineDragStart = (elID, diffInInitialLeft, diffInInitialTop, diffInFinalLeft, diffInFinalTop) => { 
+  setLineDragging(true);
+  setDiffInStartX(diffInInitialLeft);
```
Comment (omeryousaf) [inline src/Feedback.js:164]: @Shapokhan let's ask Feedback.js to only be aware of which element is being dragged and share the x,y of the point where the drag action ended. Let it be the responsibility of the element (child component) to figure out what its new start and end points should be.

Comment (Shapokhan) [reply src/Feedback.js]: @omeryousaf bhai, can you have a look at the updated code now? Calculation of updated positions logic passed to Line component. Feedback (Bechara) has no responsibility for calculation now.

**Review by omeryousaf — APPROVED** [review]
We need to review why we have to temporarily give child component's state var values to parent to keep them for a while and then return without changing them, as we are doing in onLineDragStart() and handleEvent() methods. We pass 
```
diffInInitialLeft:diffInStartX, 
diffInInitialTop:diffInStartY, 
diffInFinalLeft:diffInEndX, 
diffInFinalTop:diffInEndY
```
from Line component to Feedback (parent) component and then get these back from Feedback component via handleEvent method when it calls lineref.current.calculateUpdatedPositions().

**Pull request merged**